tests/unittests/camera_tests.py

```
@@ -0,0 +1,1461 @@
+            quaternion=[0.0, 0.0, 0.0, 1.0],
+        )
+
+        # Should not raise - frames are unspecified
+        composed = t1.compose(t2)
+        self.assertIsNone(composed.source_frame)
+        self.assertIsNone(composed.target_frame)
+
+    def test_covariance_validation(self):
+        """Test that covariance must have 6 elements."""
+        # Valid covariance
+        extrinsics = SensorExtrinsics(
```

Ruff (0.14.10)

Local variable `extrinsics` is assigned to but never used

Remove assignment to unused variable `extrinsics`

(F841)